### PR TITLE
[NO MERGE] Test Cython 0.29beta1

### PR DESCRIPTION
**Pull request by bjodah**
Beta 1 of Cython 0.29 is out:

https://groups.google.com/forum/#!topic/cython-users/qLDWiiVMJGQ

Opening this PR only to run CI against that pre-release.

### CODE REVIEW

**Comment by bjodah** [comment]
This change does not seem to introduce any new test failures.
Cython 0.29 will emit a `DeprecationWarning`:
```
/home/travis/conda_root/lib/python3.6/site-packages/Cython/Compiler/Main.py:357: FutureWarning: Cython directive 'language_level' not set, using 2 for now (Py2). This will change in a later release! File: /home/travis/build/symengine/symengine.py/symengine/lib/symengine_wrapper.pxd
```
so the language_level should probably be set.

File: cmake/FindCython.cmake [hunk 1]
```diff
@@ -5,7 +5,7 @@
 # This finds the "cython" executable in your PATH, and then in some standard
 # paths:
 SET(CYTHON_BIN cython CACHE STRING "Cython executable name")
-SET(CYTHON_FLAGS --cplus --fast-fail)
+SET(CYTHON_FLAGS --cplus --fast-fail -3)
```
Comment (certik) [inline cmake/FindCython.cmake]: This only works with Cython 0.29beta1, or does the `-3` flag also work with previous versions? I.e., whether we can enable this by default, or whether we have to check the Cython version.

Comment (bjodah) [reply cmake/FindCython.cmake]: cython-0.19.1 (minimum req. according to symengine.py's setup.py) does have the `-3` flag.

Comment (bjodah) [reply cmake/FindCython.cmake]: Looks like we should use the `-2` flag for python2 builds and `-3` flag for python3 builds, but I haven't looked into that.

**Comment by certik** [comment]
@bjodah just curious about the `-3` flag: does it mean that Cython no longer generates the same code for both Python 2.7 and Python 3.6? And so our cmake needs to figure out the Python version and then send the proper `-2` or `-3` flag?

**Comment by bjodah** [comment]
It's my understanding that the generated code will work under both Py2 & Py3, and that the change is only on how things in the `.pyx` file is interpreted such as `print(foo, end='')` (`end` being a python3 kwarg for print) & `5/7` (integer division with language_level=2 and floating point division for 3 respectively).

All errors under python2 that cropped up when changing language_level to 3 seemed to have to do with unicode literals, the `str_is_str` directive should avoid interpreting literals as unicode under Python 2.

**Comment by certik** [comment]
Ah ok --- se we should simply use Python 3 semantics in the Cython files, use `-3` and be done with it. Then just test the results under Python 2 also, and fix any issues.

**Comment by bjodah** [comment]
Yes, precisely. And looking at the Travis log it looks like all tests are passing except the previously failing sage test.

On AppVeyor however, it is failing because I used this brand new `str_is_str` directive (which will be released in 0.29):
```
Error in compiler directive: Unknown option: "str_is_str"
```
So not sure what the migration plan should be for `symengine.py` here (bumping minimum req. to cython-0.29 at some point, or conditionally set flags in FindCython.cmake based on installed cython version).

But I will report back to Cython's mailing list that the `str_is_str` directive solved the unicode literal problems we saw here when setting `language_level=3`.

**Pull request closed without merging**